relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -99,6 +215,75 @@ pub fn get_sampling_key(envelope: &Envelope) -> Option<ProjectKey> {
     envelope.dsc().map(|dsc| dsc.public_key)
 }
 
+/// Compares two semantic versions.
+///
+/// This function is just temporary since it's used to compare SDK versions when an [`Event`] is
+/// received.
+fn compare_versions(version1: &str, version2: &str) -> std::cmp::Ordering {
+    // Split the version strings into individual numbers
+    let nums1: Vec<&str> = version1.split('.').collect();
+    let nums2: Vec<&str> = version2.split('.').collect();
+
+    // Pad the shorter version with zeros to ensure equal length
+    let length = usize::max(nums1.len(), nums2.len());
+    let nums1 = {
+        let mut padded = vec!["0"; length - nums1.len()];
+        padded.extend(nums1);
+        padded
+    };
+    let nums2 = {
+        let mut padded = vec!["0"; length - nums2.len()];
+        padded.extend(nums2);
+        padded
+    };
+
+    // Compare the numbers from left to right
+    for (num1, num2) in nums1.iter().zip(nums2.iter()) {
+        let num1 = num1.parse::<i32>().unwrap_or(0);
+        let num2 = num2.parse::<i32>().unwrap_or(0);
+
+        match num1.cmp(&num2) {
+            Ordering::Greater => return Ordering::Greater,
+            Ordering::Less => return Ordering::Less,
+            Ordering::Equal => continue,
+        }
+    }
+
+    // All numbers are equal
+    Ordering::Equal
+}
+
+/// Emits a metric when an [`Event`] is inside an [`Envelope`] without [`DynamicSamplingContext`].
+///
+/// This function is a temporary function which has been added mainly for debugging purposes. Our
+/// goal with this function is to validate how many times the DSC is not
+fn track_missing_dsc(event: &Event) {
+    let Some(client_sdk_info) = event.client_sdk.value() else {
+        return;
+    };
+
+    let (Some(sdk_name), Some(sdk_version)) = (
+        client_sdk_info.name.value(),
+        client_sdk_info.version.value(),
+    ) else {
+        return;
+    };
+
+    let Some(min_sdk_version) = SUPPORTED_SDK_VERSIONS.get(sdk_name.as_str()) else {
+        return;
+    };
+
+    match compare_versions(sdk_version, min_sdk_version) {
+        Ordering::Greater | Ordering::Equal => {
+            relay_statsd::metric!(
+                counter(RelayCounters::MissingDynamicSamplingContext) += 1,
+                sdk_name = sdk_name.as_str()
```

Comment:
This tag has technically unbounded cardinality but we know that the SDK names are not many.

Reply:
You know that sdk_name is in SUPPORTED_SDK_VERSIONS, so it is bounded, right?

Reply:
Yeah in this case this is a good point, I forgot I was checking the presence in the map.